@fsx950223 can you please fix the build failures ?

`third_party/tensorflow/python/framework/test_util.py", line 2892, in _assertRaggedEqual
    self.assertEqual(a_list, b_list, msg)
AssertionError: Lists differ:`

--- a/tensorflow/python/kernel_tests/string_split_op_test.py
+++ b/tensorflow/python/kernel_tests/string_split_op_test.py
@@ -279,11 +279,11 @@ class StringSplitV2OpTest(test_util.TensorFlowTestCase, parameterized.TestCase):
       expected_sparse = self.evaluate(expected_ragged.to_sparse())
       actual_sparse_v2 = string_ops.string_split_v2(input, **kwargs)
       self.assertEqual(expected_sparse.indices.tolist(),
-                        self.evaluate(actual_sparse_v2.indices).tolist())
+                       self.evaluate(actual_sparse_v2.indices).tolist())
       self.assertEqual(expected_sparse.values.tolist(),
-                        self.evaluate(actual_sparse_v2.values).tolist())
+                       self.evaluate(actual_sparse_v2.values).tolist())
       self.assertEqual(expected_sparse.dense_shape.tolist(),
-                        self.evaluate(actual_sparse_v2.dense_shape).tolist())
+                       self.evaluate(actual_sparse_v2.dense_shape).tolist())
 
   @parameterized.named_parameters([
       {"testcase_name": "Simple",
@@ -370,16 +370,15 @@ class StringSplitV2OpTest(test_util.TensorFlowTestCase, parameterized.TestCase):
       input = ragged_factory_ops.constant(input, dtype=dtypes.string)
     else:
       input = constant_op.constant(input, dtype=dtypes.string)
-    
+
     expected_ragged = ragged_factory_ops.constant(
-    expected)
+        expected)
     actual_ragged_v1 = ragged_string_ops.strings_split_v1(
         input, result_type="RaggedTensor", **kwargs)
     actual_ragged_v1_input_kwarg = ragged_string_ops.strings_split_v1(
         input=input, result_type="RaggedTensor", **kwargs)
     actual_ragged_v1_source_kwarg = ragged_string_ops.strings_split_v1(
         source=input, result_type="RaggedTensor", **kwargs)
-    
     self.assertAllEqual(expected_ragged, actual_ragged_v1)
     self.assertAllEqual(expected_ragged, actual_ragged_v1_input_kwarg)
     self.assertAllEqual(expected_ragged, actual_ragged_v1_source_kwarg)
@@ -387,11 +386,11 @@ class StringSplitV2OpTest(test_util.TensorFlowTestCase, parameterized.TestCase):
     actual_sparse_v1 = ragged_string_ops.strings_split_v1(
         input, result_type="SparseTensor", **kwargs)
     self.assertEqual(expected_sparse.indices.tolist(),
-                      self.evaluate(actual_sparse_v1.indices).tolist())
+                     self.evaluate(actual_sparse_v1.indices).tolist())
     self.assertEqual(expected_sparse.values.tolist(),
-                      self.evaluate(actual_sparse_v1.values).tolist())
+                     self.evaluate(actual_sparse_v1.values).tolist())
     self.assertEqual(expected_sparse.dense_shape.tolist(),
-                      self.evaluate(actual_sparse_v1.dense_shape).tolist())
+                     self.evaluate(actual_sparse_v1.dense_shape).tolist())
 
   def _py_split(self, strings, **kwargs):
     if isinstance(strings, compat.bytes_or_text_types):

--- a/tensorflow/python/kernel_tests/string_split_op_test.py
+++ b/tensorflow/python/kernel_tests/string_split_op_test.py
@@ -267,18 +267,9 @@ class StringSplitV2OpTest(test_util.TensorFlowTestCase, parameterized.TestCase):
     # correctly.
     expected_ragged = ragged_factory_ops.constant(
         expected, ragged_rank=input.shape.ndims)
-    actual_ragged_v1 = ragged_string_ops.strings_split_v1(
-        input, result_type="RaggedTensor", **kwargs)
-    actual_ragged_v1_input_kwarg = ragged_string_ops.strings_split_v1(
-        input=input, result_type="RaggedTensor", **kwargs)
-    actual_ragged_v1_source_kwarg = ragged_string_ops.strings_split_v1(
-        source=input, result_type="RaggedTensor", **kwargs)
     actual_ragged_v2 = ragged_string_ops.string_split_v2(input, **kwargs)
     actual_ragged_v2_input_kwarg = ragged_string_ops.string_split_v2(
         input=input, **kwargs)
-    self.assertAllEqual(expected_ragged, actual_ragged_v1)
-    self.assertAllEqual(expected_ragged, actual_ragged_v1_input_kwarg)
-    self.assertAllEqual(expected_ragged, actual_ragged_v1_source_kwarg)
     self.assertAllEqual(expected_ragged, actual_ragged_v2)
     self.assertAllEqual(expected_ragged, actual_ragged_v2_input_kwarg)
 
@@ -286,16 +277,121 @@ class StringSplitV2OpTest(test_util.TensorFlowTestCase, parameterized.TestCase):
     # correctly.  Note: the internal version oly supports vector inputs.
     if input.shape.ndims == 1:
       expected_sparse = self.evaluate(expected_ragged.to_sparse())
-      actual_sparse_v1 = ragged_string_ops.strings_split_v1(
-          input, result_type="SparseTensor", **kwargs)
       actual_sparse_v2 = string_ops.string_split_v2(input, **kwargs)
-      for actual_sparse in [actual_sparse_v1, actual_sparse_v2]:
-        self.assertEqual(expected_sparse.indices.tolist(),
-                         self.evaluate(actual_sparse.indices).tolist())
-        self.assertEqual(expected_sparse.values.tolist(),
-                         self.evaluate(actual_sparse.values).tolist())
-        self.assertEqual(expected_sparse.dense_shape.tolist(),
-                         self.evaluate(actual_sparse.dense_shape).tolist())
+      self.assertEqual(expected_sparse.indices.tolist(),
+                        self.evaluate(actual_sparse_v2.indices).tolist())
+      self.assertEqual(expected_sparse.values.tolist(),
+                        self.evaluate(actual_sparse_v2.values).tolist())
+      self.assertEqual(expected_sparse.dense_shape.tolist(),
+                        self.evaluate(actual_sparse_v2.dense_shape).tolist())
+
+  @parameterized.named_parameters([
+      {"testcase_name": "Simple",
+       "input": [b"pigs on the wing", b"animals"],
+       "expected": [[b"pigs", b"on", b"the", b"wing"], [b"animals"]]},
+
+      {"testcase_name": "MultiCharSeparator",
+       "input": [b"1<>2<>3", b"<><>4<>5<><>6<>"],
+       "sep": b"<>",
+       "expected": [[b"1", b"2", b"3"],
+                    [b"", b"", b"4", b"5", b"", b"6", b""]]},
+
+      {"testcase_name": "SimpleSeparator",
+       "input": [b"1,2,3", b"4,5,,6,"],
+       "sep": b",",
+       "expected": [[b"1", b"2", b"3"], [b"4", b"5", b"", b"6", b""]]},
+
+      {"testcase_name": "EmptySeparator",
+       "input": [b"1 2 3", b"  4  5    6  "],
+       "expected": [[b"1", b"2", b"3"], [b"4", b"5", b"6"]]},
+
+      {"testcase_name": "EmptySeparatorEmptyInputString",
+       "input": [b""],
+       "expected": [[]]},
+
+      {"testcase_name": "SimpleSeparatorMaxSplit",
+       "input": [b"1,2,3", b"4,5,,6,"],
+       "sep": b",",
+       "maxsplit": 1,
+       "expected": [[b"1", b"2,3"], [b"4", b"5,,6,"]]},
+
+      {"testcase_name": "EmptySeparatorMaxSplit",
+       "input": [b"1 2 3", b"  4  5    6  "],
+       "maxsplit": 1,
+       "expected": [[b"1", b"2 3"], [b"4", b"5    6  "]]},
+
+      {"testcase_name": "ScalarInput",
+       "input": b"1,2,3",
+       "sep": b",",
+       "expected": [[b"1", b"2", b"3"]]},
+
+      {"testcase_name": "Dense2DInput",
+       "input": [[b"1,2,3", b"4"], [b"5,6", b"7,8,9"]],
+       "sep": b",",
+       "expected": [[[b"1", b"2", b"3"], [b"4"]],
+                    [[b"5", b"6"], [b"7", b"8", b"9"]]]},
+
+      {"testcase_name": "Ragged2DInput",
+       "input": [[b"1,2,3", b"4"], [b"5,6"]],
+       "input_is_ragged": True,
+       "sep": b",",
+       "expected": [[[b"1", b"2", b"3"], [b"4"]], [[b"5", b"6"]]]},
+
+      {"testcase_name": "Ragged3DInput",
+       "input": [[[b"1,2,3", b"4"], [b"5,6"]], [[b"7,8,9"]]],
+       "input_is_ragged": True,
+       "sep": b",",
+       "expected": [[[[b"1", b"2", b"3"], [b"4"]], [[b"5", b"6"]]],
+                    [[[b"7", b"8", b"9"]]]]},
+
+      {"testcase_name": "Ragged4DInput",
+       "input": [[[[b"1,2,3", b"4"], [b"5,6"]], [[b"7,8,9"]]], [[[b""]]]],
+       "input_is_ragged": True,
+       "sep": b",",
+       "expected": [[[[[b"1", b"2", b"3"], [b"4"]], [[b"5", b"6"]]],
+                     [[[b"7", b"8", b"9"]]]], [[[[b""]]]]]},
+
+      {"testcase_name": "Ragged4DInputEmptySeparator",
+       "input": [[[[b"1 2 3", b"4"], [b"5 6"]], [[b"7 8 9"]]], [[[b""]]]],
+       "input_is_ragged": True,
+       "expected": [[[[[b"1", b"2", b"3"], [b"4"]], [[b"5", b"6"]]],
+                     [[[b"7", b"8", b"9"]]]], [[[[]]]]]},
+
+      ])  # pyformat: disable
+  def testSplitV1(self,
+                  input,
+                  expected,
+                  input_is_ragged=False,
+                  **kwargs):
+    # self.assertEqual(expected, self._py_split(input, **kwargs))
+
+    # Prepare the input tensor.
+    if input_is_ragged:
+      input = ragged_factory_ops.constant(input, dtype=dtypes.string)
+    else:
+      input = constant_op.constant(input, dtype=dtypes.string)
+    
+    expected_ragged = ragged_factory_ops.constant(
+    expected)
+    actual_ragged_v1 = ragged_string_ops.strings_split_v1(
+        input, result_type="RaggedTensor", **kwargs)
+    actual_ragged_v1_input_kwarg = ragged_string_ops.strings_split_v1(
+        input=input, result_type="RaggedTensor", **kwargs)
+    actual_ragged_v1_source_kwarg = ragged_string_ops.strings_split_v1(
+        source=input, result_type="RaggedTensor", **kwargs)
+    
+    self.assertAllEqual(expected_ragged, actual_ragged_v1)
+    self.assertAllEqual(expected_ragged, actual_ragged_v1_input_kwarg)
+    self.assertAllEqual(expected_ragged, actual_ragged_v1_source_kwarg)
+    expected_sparse = self.evaluate(expected_ragged.to_sparse())
+    actual_sparse_v1 = ragged_string_ops.strings_split_v1(
+        input, result_type="SparseTensor", **kwargs)
+    self.assertEqual(expected_sparse.indices.tolist(),
+                      self.evaluate(actual_sparse_v1.indices).tolist())
+    self.assertEqual(expected_sparse.values.tolist(),
+                      self.evaluate(actual_sparse_v1.values).tolist())
+    self.assertEqual(expected_sparse.dense_shape.tolist(),
+                      self.evaluate(actual_sparse_v1.dense_shape).tolist())
 
   def _py_split(self, strings, **kwargs):
     if isinstance(strings, compat.bytes_or_text_types):

--- a/tensorflow/python/kernel_tests/string_split_op_test.py
+++ b/tensorflow/python/kernel_tests/string_split_op_test.py
@@ -362,9 +362,7 @@ class StringSplitV2OpTest(test_util.TensorFlowTestCase, parameterized.TestCase):
                   input,
                   expected,
                   input_is_ragged=False,
-                  **kwargs):
-    # self.assertEqual(expected, self._py_split(input, **kwargs))
-
+                  **kwargs):  # pylint: disable=redefined-builtin
     # Prepare the input tensor.
     if input_is_ragged:
       input = ragged_factory_ops.constant(input, dtype=dtypes.string)

--- a/tensorflow/python/ops/ragged/ragged_string_ops.py
+++ b/tensorflow/python/ops/ragged/ragged_string_ops.py
@@ -629,7 +629,7 @@ def strings_split_v1(input=None, sep=None, maxsplit=-1,  # pylint: disable=redef
     input = ragged_tensor.convert_to_tensor_or_ragged_tensor(
         input, dtype=dtypes.string, name="input")
 
-    if result_type == "SparseTensor" and input.shape.rank==0:
+    if input.shape.rank == 0:
       input = gen_array_ops.expand_dims(input, 0)
     ragged_result = string_split_v2(input, sep=sep, maxsplit=maxsplit)
     if result_type == "SparseTensor":

